frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import { withRouter } from "react-router";
+import { connect } from "react-redux";
+import cx from "classnames";
+import _ from "underscore";
+import type { Location } from "history";
+
+import TitleAndDescription from "metabase/components/TitleAndDescription";
+
+import { useOnMount } from "metabase/hooks/use-on-mount";
+import { getSetting } from "metabase/selectors/settings";
+import { IFRAMED, initializeIframeResizer } from "metabase/lib/dom";
+import { parseHashOptions } from "metabase/lib/browser";
+
+import SyncedParametersList from "metabase/parameters/components/SyncedParametersList/SyncedParametersList";
+
+import type { Dashboard, Parameter, ParameterId } from "metabase-types/api";
+import type { ParameterValueOrArray } from "metabase-types/types/Parameter";
+import type { State } from "metabase-types/store";
+
+import { getValuePopulatedParameters } from "metabase-lib/parameters/utils/parameter-values";
+
+import LogoBadge from "./LogoBadge";
+import {
+  Root,
+  ContentContainer,
+  Header,
+  Body,
+  Footer,
+  ActionButtonsContainer,
+} from "./EmbedFrame.styled";
+import "./EmbedFrame.css";
+
+interface OwnProps {
+  className?: string;
+  name?: string;
+  description?: string;
+  dashboard?: Dashboard;
+  actionButtons?: JSX.Element[];
+  parameters?: Parameter[];
+  parameterValues?: Record<ParameterId, ParameterValueOrArray>;
+  setParameterValue?: (parameterId: ParameterId, value: any) => void;
+  children: React.ReactNode;
+}
+
+interface StateProps {
+  hasEmbedBranding: boolean;
+}
+
+type Props = OwnProps &
+  StateProps & {
+    location: Location;
+  };
+
+interface HashOptions {
+  bordered?: boolean;
+  titled?: boolean;
+  theme?: string;
+  hide_parameters?: string;
+  hide_download_button?: boolean;
+}
+
+function mapStateToProps(state: State) {
+  return {
+    hasEmbedBranding: !getSetting(state, "hide-embed-branding?"),
+  };
+}
+
+function EmbedFrame({
+  className,
+  children,
+  name,
+  description,
+  dashboard,
+  actionButtons,
+  location,
+  hasEmbedBranding,
+  parameters,
+  parameterValues,
+  setParameterValue,
+}: Props) {
+  const [hasInnerScroll, setInnerScroll] = useState(true);
+
+  useOnMount(() => {
+    initializeIframeResizer(() => setInnerScroll(false));
+  });
+
+  const {
+    bordered = IFRAMED,
+    titled = true,
+    theme,
+    hide_parameters,
+    hide_download_button,
+  } = parseHashOptions(location.hash) as HashOptions;
+
+  const showFooter =
+    hasEmbedBranding || (!hide_download_button && actionButtons);
+
+  const finalName = titled ? name : null;
+
+  const hasParameters =
+    Array.isArray(parameters) &&
+    parameters.length > 0 &&
+    typeof setParameterValue === "function";
+
+  const hasHeader = Boolean(finalName || hasParameters);
```

Comment:
Technically, this isn't a 1:1 refactoring from the previous code. If we pass `name`, and `parameters`, but not `setParameterValue`.

- In the previous version, we'll render `<TitleAndDescription />`
- This version, we'll **not** render `<TitleAndDescription />` since the absence of `setParameterValue` makes `hasParameters = false`

Maybe it's better to inline `&& setParameterValue` at line 126 instead so it looks like this
```ts
{hasParameters && setParameter && (...
```

P.S. I believe the `setParameter` check is there to prevent TypeScript from complaining.

Reply:
Great catch, thanks! The trouble is that `SyncedParametersList` always expects `setParameterValue` to be passed. However, the only thing it does with it, is that it passes it down to `ParametersList` that can handle `undefined` `setParameterValue` (see [here](https://github.com/metabase/metabase/blob/2553505eb8775a8a90273b993fcdfbd194bd4bed/frontend/src/metabase/parameters/components/ParametersList.jsx#L105)). I've removed `setParameterValue` from `hasParameters` check